### PR TITLE
MMT-2710: update UMM-C to v1.16.6

### PR DESCRIPTION
**Pull request by charleshuang80**
* copy and merge UMM-C v1.16.6
* add FreeAndOpenData field to UseConstraints fields
* modify UseConstraints type radio button selection and behavior
* modify clear radio button link data attrs and update function
* update validation for UseConstraints fields with oneOf and new field behavior
* update factory
* update tests
* add migration

### CODE REVIEW

File: app/views/collection_drafts/forms/fields/_geometry_fields.html.erb [hunk 1]
```diff
@@ -32,8 +32,8 @@
       prefix: name_prefix,
       required: true
     ) %>
-    <a href="javascript:void(0)" class="clear-radio-button" data-content="coordinate-system-type">Clear</a>
-    <div class="coordinate-system-type-group">
+    <a href="javascript:void(0)" class="clear-radio-button" data-clear-fields="coordinate-system-type-clear-link-clear-fields" data-hide-fields="coordinate-system-type-clear-link-hide-fields" data-radio-fields="coordinate-system-type-clear-link-radio-fields">Clear</a>
```
Comment (cgokey) [inline app/views/collection_drafts/forms/fields/_geometry_fields.html.erb:35]: should this say: resolution-and-coordinate-system-type to be consistent?

File: app/views/collection_drafts/forms/fields/_geometry_fields.html.erb [hunk 2]
```diff
@@ -32,8 +32,8 @@
...
+    <div class="coordinate-system-type-group coordinate-system-type-clear-link-radio-fields">
```
Comment (cgokey) [inline app/views/collection_drafts/forms/fields/_geometry_fields.html.erb:36]: likewise here

File: spec/factories/collection_factory_data.rb [hunk 3]
```diff
@@ -594,6 +594,7 @@ def collection_one
     'Quality'            => 'Metadata quality summary',
     'UseConstraints' => {
       'Description' => 'These are some use constraints',
+      'FreeAndOpenData' => true,
```
Comment (cgokey) [inline spec/factories/collection_factory_data.rb:597]: Is this correct?   Shouldn't this have a description and/or type?

Comment (charleshuang80) [reply spec/factories/collection_factory_data.rb:597]: The type is not an actual field, the radio buttons are just so the form knows what fields to show or hide.
The description is just above.

File: app/views/collection_drafts/forms/fields/_geometry_fields.html.erb (hunk 1 above)
Comment (charleshuang80) [reply app/views/collection_drafts/forms/fields/_geometry_fields.html.erb:35]: So this is the reason I changed the other radio buttons to `resolution-and-coordinate-system-type`. This is actually correct and for a different field. So the other was kind of a dupe and made for confusion.

**Review by cgokey — APPROVED** [review]

**Review by htranho — APPROVED** [review]

**Pull request merged**